### ISSUE TITLE
rndemo not working

### ISSUE THREAD

**Issue by Letty** [acro5piano/react-native-big-calendar#922]
I've checked out your repo and tried to get the demo running based on the readme.
When running `yarn start` it will produce an error:
```
Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at stableHash (/Users/.../react-native-big-calendar/rndemo/node_modules/metro-cache/src/stableHash.js:19:8)
    at Object.getCacheKey (/Users/.../react-native-big-calendar/rndemo/node_modules/metro-transform-worker/src/index.js:647:7)
    at getTransformCacheKey (/Users/.../react-native-big-calendar/rndemo/node_modules/metro/src/DeltaBundler/getTransformCacheKey.js:32:19)
    at new Transformer (/Users/.../react-native-big-calendar/rndemo/node_modules/metro/src/DeltaBundler/Transformer.js:180:28)
    at /Users/.../react-native-big-calendar/rndemo/node_modules/metro/src/Bundler.js:58:29
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

and `yarn start-rndemo-android` gives: `error Command "start-rndemo-android" not found.`

Before that i also tried running it via XCode on iOS, but i think in the readme is one step missing for iOS development: `npx pod-install`. But its also not running on iOS.

Maybe i can update the demo to the latest react-native version and this will solve the issue?

**Comment by acro5piano** [comment]
@Letty Thank you! The document for demo has not been updated recently.

> and yarn start-rndemo-android gives: error Command "start-rndemo-android" not found.
> Before that i also tried running it via XCode on iOS, but i think in the readme is one step missing for iOS development: npx pod-install. But its also not running on iOS.

Thanks for pointing out it. I've fixed them in this commit: https://github.com/acro5piano/react-native-big-calendar/commit/842639c893c5f54bfd044dbbf82d5acc6ac9afcf

> Maybe i can update the demo to the latest react-native version and this will solve the issue?

I hope so. If you succeed to update it, please send a PR to this library; that would be helpful.
I recommend creating a whole new RN  project with the latest RN version and install this library then update code, because updating RN version is really painful.

**Comment by Letty** [comment]
> I hope so. If you succeed to update it, please send a PR to this library; that would be helpful. I recommend creating a whole new RN project with the latest RN version and install this library then update code, because updating RN version is really painful.

I will give it a try!

**Comment by Letty** [comment]
The demo is updated and uses the latest react native version.
One thing I changed is the Picker. This Element is no longer a part of rn and I simply put some buttons to change the mode.

One problem i had during the commit was an error in the pre-commit hook. I guess it is some sort of an issue with the config files in the rndemo. The error was: `Parsing error: No Babel config file detected`. I bypassed it with removing the `lint-staged` config in the package.json. I hope that was alright. Not sure which config needs to be adapted to avoid this error.

**Comment by acro5piano** [comment]
Implemented on 4.4.0.

> One problem i had during the commit was an error in the pre-commit hook. I guess it is some sort of an issue with the config files in the rndemo. The error was: Parsing error: No Babel config file detected. I bypassed it with removing the lint-staged config in the package.json. I hope that was alright. Not sure which config needs to be adapted to avoid this error.

Sorry for taking your time in the hook. I'll take a look into the issue next time I modify the demo app.
